components/organisms/UserSettingsPage/user-settings-page.tsx

```diff
@@ -33,24 +33,44 @@ const UserSettingsPage = ({ user }: userSettingsPageProps) => {
   const [displayLocalTime, setDisplayLocalTime] = useState(false);
   const [timezone, setTimezone] = useState("");
   const [userInfo, setUserInfo] = useState<DbUser>();
-  const [email, setEmail] = useState<string | undefined>(userInfo?.email || user?.email);
+  const [email, setEmail] = useState<string | undefined>("");
   const [emailPreference, setEmailPreference] = useState<EmailPreferenceType>({
     // eslint-disable-next-line camelcase
     display_email: false,
     // eslint-disable-next-line camelcase
     receive_collaboration: false
   });
   const [selectedInterest, setSelectedInterest] = useState<string[]>([]);
+  const nameRef = React.useRef<HTMLInputElement>(null);
+  const bioRef = React.useRef<HTMLTextAreaElement>(null);
+  const urlRef = React.useRef<HTMLInputElement>(null);
+  const twitterRef = React.useRef<HTMLInputElement>(null);
+  const companyRef = React.useRef<HTMLInputElement>(null);
+  const locationRef = React.useRef<HTMLInputElement>(null);
   const interestArray = ["javascript", "python", "rust", "ML", "AI", "react"];
 
   useEffect(() => {
     async function fetchAuthSession() {
       const response = await authSession();
-      if (response !== false) setUserInfo(response);
+      if (response !== false) {
+        setUserInfo(response);
+        nameRef.current!.value = response.name;
+        setEmail(response.email); // getting email from response instead of user.email
+        bioRef.current!.value = response.bio;
+        urlRef.current!.value = response.url;
+        twitterRef.current!.value = response.twitter_username;
+        companyRef.current!.value = response.company;
+        locationRef.current!.value = response.location;
+      }
     }
+    // if (user) setEmail(user.email);
+    fetchAuthSession();
+  }, [user]);
 
-    if (user) setEmail(user.email);
-    if (insightsUser) {
+  // insightsUsers uses SWR which keeps triggering re-rendering and overwriting the user input. Disabling this for now
+  let firstInsightsLoad = false;
+  useEffect(() => {
+    if (insightsUser && !firstInsightsLoad) {
```

Comment:
I'd prefer this to be fixed in the hook if possible to not keep refreshing the data.

Reply:
Turned out this is actually related to user object being re-assigned frequently from `useSupabaseAuth` hook. Should be addressed in a new issue.

Reply:
Or I can remove `user` from the useEffect dependency array if that won't cause side problems